python/dgl/heterograph.py

```diff
@@ -5251,4 +4378,72 @@ def __repr__(self):
             return ret.format(
                 srcnode=nsrcnode_dict, dstnode=ndstnode_dict, edge=nedge_dict, meta=meta)
 
+
+def _create_compute_graph(graph, u, v, eid, recv_nodes=None):
+    """Create a computation graph from the given edges.
+
+    The compute graph is a uni-directional bipartite graph with only
+    one edge type. There is no orphan destination nodes, i.e., all the
+    dst nodes have at least one in-coming edge.
+
+    Similar to subgraph extraction, it stores the original node IDs
+    in the srcdata[NID] and dstdata[NID] and extracts features accordingly.
+    Edges are not relabeled.
+
+    This function is typically used during message passing to generate
+    a graph that contains only the active set of edges.
+
+    Parameters
+    ----------
+    graph : DGLGraph
+        The input graph.
+    u : Tensor
+        Src nodes.
+    v : Tensor
+        Dst nodes.
+    eid : Tensor
+        Edge IDs.
+    recv_nodes : Tensor
+        Nodes that receive messages. If None, it is equal to unique(v).
+        Otherwise, it must be a super set of v and can contain nodes
```

Comment:
"super set" -> "superset".

Also, if the dst nodes must have at least one incoming edge, can `recv_nodes` still be a superset of `v`?

Reply:
It could but there is no such case. The function is called by `send_and_recv`, `pull` and `update_all`. Only `pull` will set the `recv_nodes` and could contain zero-degree nodes.

Reply:
If there is the case when `recv_nodes` contains zero-degree nodes, then the description `There is no orphan destination nodes...` above should be removed.

Reply:
done